### ISSUE TITLE
Get Involved

### ISSUE THREAD

**Issue by Shoalsteed** [Shoalsteed/I2P-Secure-Design-Collective#66]
A list of specific tasks to point to when people ask how they can help. This should also be viewed as a way to onboard new long term contributors and ensure that the team has up to date learning resources for new devs.

**Comment by Shoalsteed** [comment]
Each task or contribution item should have:
- clear scope 
- guides for onboarding
- contact person

**Comment by eyedeekay** [comment]
Realistic Example: Replace the Easy-Install for Mac Signer

 - Contributors to the I2P Easy-Install for Mac. Contributors who demonstrate responsibility by virtue of their contributions could eventually be considered for maintainership of the bundle, if they are willing to also become the signer of the bundle.
  - Experience in Java Development
  - Experience in Apple Application Development
  - Should be able to complete: https://github.com/i2p/i2p-jpackage-mac/blob/master/BUILD.md successfully
 - Contact us by filing issues against https://i2pgit.org/i2p-hackers/i2p-jpackage-mac

**Comment by Shoalsteed** [comment]
**Support I2P Core Development**
The project hosts meetings on the first Tuesday of every month that are open to the community. This is a great opportunity to see what is happening with I2P core development and familiarize yourself with Roadmap tasks. Additionally, protocol development meetings take place every Tuesday and zzz's development forum is available all of the time as a place to participate in development and issues that need fixing. See the new developer's guide for how to get started.

**Help translate the website and the software into your language**
 Translators are a very important part of the project and your work is always appreciated. See the [new translator's guide](https://geti2p.net/en/get-involved/guides/new-translators) for details. The project also welcomes support to help keep its documentation updated. See the new translator's guide for details.

**Host a Reseed Server**
 Helping new users onto the network is a very important task, and that task is handled by reseed servers. The more reseed servers the network has, the more de-centralized and redundant its infrastructure is. Detailed instructions can be found in the [reseed server page](https://geti2p.net/en/get-involved/guides/reseed).

**Create Applications For the I2P Network**
 Write or port applications for I2P. There's some guidelines and a list of ideas on the [applications page](https://geti2p.net/en/get-involved/develop/applications).

**Test and Review Code**
Run test builds and provide feedback during release cycles. 

**Research and Security** 
Help the project find where it may need to address a vulnerability. Study or test the code to look for vulnerabilities. Both anonymity vulnerabilities from the various [threat models](https://geti2p.net/en/docs/how/threat-model), and DOS and other weaknesses due to security holes, benefit from ongoing research

**Comment by Shoalsteed** [comment]
> Realistic Example: Get contributors for Easy-Install Mac, replace the Easy-Install for Mac Lead
> 
> * Contributors to the I2P Easy-Install for Mac. Contributors who demonstrate responsibility by virtue of their contributions could eventually be considered for maintainership of the bundle, if they are willing to also become the signer of the bundle.
> * Experience in Java Development
> * Experience in Apple Application Development
> * Should be able to complete: https://github.com/i2p/i2p-jpackage-mac/blob/master/BUILD.md successfully
> * Contact us by filing issues against https://i2pgit.org/i2p-hackers/i2p-jpackage-mac

Yes - this includes scope ,  guidance for a new contributor and a specific task.

**Comment by Shoalsteed** [comment]
> **Support I2P Core Development** The project hosts meetings on the first Tuesday of every month that are open to the community. This is a great opportunity to see what is happening with I2P core development and familiarize yourself with Roadmap tasks. Additionally, protocol development meetings take place every Tuesday and zzz's development forum is available all of the time as a place to participate in development and issues that need fixing. See the new developer's guide for how to get started.
> 
> **Help translate the website and the software into your language** Translators are a very important part of the project and your work is always appreciated. See the [new translator's guide](https://geti2p.net/en/get-involved/guides/new-translators) for details. The project also welcomes support to help keep its documentation updated. See the new translator's guide for details.
> 
> **Host a Reseed Server** Helping new users onto the network is a very important task, and that task is handled by reseed servers. The more reseed servers the network has, the more de-centralized and redundant its infrastructure is. Detailed instructions can be found in the [reseed server page](https://geti2p.net/en/get-involved/guides/reseed).
> 
> **Create Applications For the I2P Network** Write or port applications for I2P. There's some guidelines and a list of ideas on the [applications page](https://geti2p.net/en/get-involved/develop/applications).
> 
> **Test and Review Code** Run test builds and provide feedback during release cycles.
> 
> **Research and Security** Help the project find where it may need to address a vulnerability. Study or test the code to look for vulnerabilities. Both anonymity vulnerabilities from the various [threat models](https://geti2p.net/en/docs/how/threat-model), and DOS and other weaknesses due to security holes, benefit from ongoing research

THIS is what is salvageable from the existing page, however the copy needs to be improved.

**Comment by Shoalsteed** [comment]
**Participate In Maintaining I2P Development**

- Experience in Java Development
  - Experience in Windows and macOS Application Development
  - Participate in testing during our release cycle
  - Should be able to complete: https://github.com/i2p/i2p-jpackage-mac/blob/master/BUILD.md successfully
 - Contact us by filing issues against 
 - https://i2pgit.org/i2p-hackers/i2p-jpackage-mac
 - https://i2pgit.org/i2p-hackers/i2p.firefox 

Contributors should be able to participate in I2P Dev monthly meetings.

**Comment by eyedeekay** [comment]
Much better, but it should also have http://i2pgit.org/i2p-hackers/i2p.firefox in contact filing issues against if we're doing both Windows and OSX in this entry

**Comment by Shoalsteed** [comment]
My next question is about the reseed process. 
As I understand it, we have no formal meetings for reseed operators. Aside from the docs on the site, is there a place for reseed operators to connect?

**Comment by Shoalsteed** [comment]
> Much better, but it should also have http://i2pgit.org/i2p-hackers/i2p.firefox in contact filing issues against if we're doing both Windows and OSX in this entry

and the build linked - is that the only resource required?

**Comment by Shoalsteed** [comment]
I am also going to include my own asks for the Secure Design Collective

**Comment by eyedeekay** [comment]
No the Windows one is https://github.com/i2p/i2p.firefox/blob/master/README.md under "End-to-End Windows Build Process using WSL"

**Comment by eyedeekay** [comment]
And it's way harder than the Mac version of the same thing unfortunately

**Comment by eyedeekay** [comment]
- Contributors who want to work on the I2P code should be able to connect to Irc2P and should join the #i2p-dev channel, which is moderated. They must register a nickname on Irc2P to use #i2p-dev. 
- They should be familiar with the basics of using SSH and be able to sign and encrypt email with GPG
- Git experience is helpful but not required

re: top one, the guide I used was here: http://deadlytechnology.com/linux/irc-on-ubuntu/ or we could write our own

**Comment by Shoalsteed** [comment]
> http://deadlytechnology.com/linux/irc-on-ubuntu/

the link goes to a suspended page for me - is it for how to register a nick name?

**Comment by eyedeekay** [comment]
Shoot, I just pulled it out of my bookmarks and didn't check if it was still up

**Comment by eyedeekay** [comment]
Yes it's for registering a nick

**Comment by eyedeekay** [comment]
https://libera.chat/guides/registration

**Comment by Shoalsteed** [comment]
> https://libera.chat/guides/registration

Is there a place to register a nick in irc2p?

**Comment by eyedeekay** [comment]
> > https://libera.chat/guides/registration
> 
> Is there a place to register a nick in irc2p?

Yes it works the same way as the other nickservers, you send a message to NickServe containing the name and password

**Comment by Shoalsteed** [comment]
{% extends "global/layout.html" %} {% block title %}{% trans %}Get Involved{% endtrans %}{% endblock %} {% block content %}

   <li><b>{% trans %}Participate In Maintaining I2P Core Development{% endtrans %}</b>{% trans %}The project hosts meetings on the first Tuesday of every month that are open to join. This is a great opportunity to see what is happening with I2P core development and
       get familiar with Roadmap tasks. Additionally, protocol development meetings take place every Tuesday on irc #ls2. Contributors who want to work on the I2P code should be able to connect to Irc2P and should join the #i2p-dev channel, which is moderated. They must register a nickname on Irc2P to use #i2p-dev. Experience in Java development
  and in Windows and macOS application development is essential for contributors who want to support core software releases. Understanding the basics of using SSH and signing and encrypting email with GPG is helpful as well.{% endtrans %}
   </li>
    <li><b><a href="{{ newtrans }}">{{ _('Translation and Documentation') }}</a></b> &mdash; {% trans newtrans=site_url('get-involved/guides/new-translators') -%} Help translate the website and the software. See the new translator's guide for details. {%- endtrans %}
    </li>
</ul>
<h3>{% trans %}Host Services for Yourself and Others{% endtrans %}</h3>
<ul>
    <li><b><a href="{{ site_url('/about/software') }}">{{ _('Services') }}</a></b> &mdash; {% trans -%} Self-hosting almost anything, from an SSH server for yourself to an ActivityPub forum for everyone and anything in between, is helpful to the I2P network,
        especially if you write down instructions for others. Almost anything you can think of can be made to work with I2P, and your service is valuable to the network. {%- endtrans %}
    </li>
    <li><b><a href="{{ reseed }}">{{ _('Reseeding') }}</a></b> &mdash; {% trans reseed=site_url('get-involved/guides/reseed') -%} Bootstrapping new routers onto the network is a very important task, and that task is handled by reseed servers. The more reseed
        servers the I2P network has, the more decentralized and redundant its infrastructure is. Detailed instructions can be found on the Reseed Server page <a href="{{ reseed }}">reseed server page</a>.
        {%- endtrans %}
    </li>
</ul>
<h3>{% trans %}Develop and Test Software{% endtrans %}</h3>
<ul>
    <li><b><a href="{{ site_url('/get-involved/guides/dev-guidelines') }}">{{ _('Applications') }}</a></b> &mdash; {% trans apps=site_url('get-involved/develop/applications') -%} Write or port applications for I2P. There's some guidelines and a list of ideas
        on the <a href="{{ apps }}">applications page</a>. {%- endtrans %}
    </li>
    <li><a href="{{ site_url('/get-involved/guides/new-developers') }}"><b>{{ _('Coding') }}</a></b> &mdash; {% trans trac=i2pconv('trac.i2p2.i2p'), zzz=i2pconv('zzz.i2p'), newdevs=site_url('get-involved/guides/new-developers') -%} Check for open tickets on Gitlab or the TODO list on <a href="http://{{ zzz }}">{{ zzz }}</a> for some ideas on where to start. See the <a href="{{ newdevs }}">new developer's guide</a>        for details. {%- endtrans %}
    </li>
    <li><b><a href="{{ site_url('research/index') }}">{{ _('Analysis') }}</b> &mdash; {% trans threatmodel=site_url('docs/how/threat-model') -%} Study and review I2P code. I2P network security and the protection of its participants benefits from ongoing research. {%- endtrans %}
    </li>
</ul>
<h4>{% trans %}Join I2P On Gitlab{% endtrans %}</h4>
<ul>
    <li><strong><a href="http://git.idk.i2p">{% trans %}Inside I2P - (http://git.idk.i2p){% endtrans %}</a></strong>
    </li>
    <li><strong><a href="https://i2pgit.org">{% trans %}Outside I2P - (https://i2pgit.org){% endtrans %}</a></strong>
    </li>
</ul>
{% endblock %}

**Comment by Shoalsteed** [comment]
file with updates posted to Gitlab for review.
